### PR TITLE
Add url host and protocol in mediawiki class

### LINKED ISSUE #1160: webUrlHost webUrlProtocol should be a property of Mediawiki class
Everywhere the code should rely on `mw.webUrlHost()`. It should not be (re-)computed each time (so lazy property) locally when we use it. It should return a value computed at the time of `mw.webUrl()` has been set.

Same with `urlParser.parse(mw.webUrl).protocol`

### PR DESCRIPTION
**Pull request by bakshiutkarsha**
should resolve #1160

### CODE REVIEW

File: src/MediaWiki.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ class MediaWiki {
   public readonly modulePath: string;
   public readonly spaceDelimiter: string;
   public readonly webUrl: string;
+  public readonly webUrlHost: string;
+  public readonly protocol: string;
```
Comment (kelson42) [inline src/MediaWiki.ts:19]: This should be called `webUrlProtocol`.

File: src/util/rewriteUrls.ts [hunk 2]
```diff
@@ -46,8 +45,7 @@ export async function rewriteUrl(articleId: string, mw: MediaWiki, dump: Dump, l
     }
 
     if (!hrefProtocol && href.slice(0, 2) === '//') {
-        const wikiProtocol = urlParser.parse(mw.webUrl).protocol;
-        href = `${wikiProtocol}${href}`;
+        href = `${mw.protocol}${href}`;
         linkNode.setAttribute('href', href);
```
Comment (kelson42) [inline src/util/rewriteUrls.ts:49]: Remove `href` variable and just assign `${mw.protocol}${href}`

File: src/util/rewriteUrls.ts [hunk 3]
```diff
@@ -46,8 +45,7 @@ export async function rewriteUrl(articleId: string, mw: MediaWiki, dump: Dump, l
...
         hrefProtocol = urlParser.parse(href).protocol;
```
Comment (kelson42) [inline src/util/rewriteUrls.ts:50]: `hrefProtocol = mw.protocol` seems better to me.

File: src/util/saveArticles.ts [hunk 4]
```diff
@@ -424,7 +423,7 @@ export async function treatVideo(mw: MediaWiki, dump: Dump, srcCache: KVS<boolea
 
     /* Scrape subtitle */
     for (const track of Array.from(videoEl.querySelectorAll('track'))) {
-        subtitles.push(await treatSubtitle(track, webUrlHost, mw, articleId));
+        subtitles.push(await treatSubtitle(track, mw.webUrlHost, mw, articleId));
```
Comment (kelson42) [inline src/util/saveArticles.ts:426]: I think now `treatSubtitle` would be advantageously changed to `treatSubtitle(track, mw, articleId)`?!

File: src/util/rewriteUrls.ts (hunk 2 above)
Comment (bakshiutkarsha) [reply src/util/rewriteUrls.ts:49]: The reason of having the `href` variable is it is getting used in other places as well.

**Pull request merged**